python-checks/src/test/java/org/sonar/python/checks/PandasDataFrameToNumpyCheckTest.java

```diff
@@ -20,13 +20,81 @@
 package org.sonar.python.checks;
 
 import org.junit.jupiter.api.Test;
+import org.sonar.python.checks.quickfix.PythonQuickFixVerifier;
 import org.sonar.python.checks.utils.PythonCheckVerifier;
 
 class PandasDataFrameToNumpyCheckTest {
 
+  PandasDataFrameToNumpyCheck check = new PandasDataFrameToNumpyCheck();
+
+  private static final String QUICK_FIX_MESSAGE = "Replace this with a call to DataFrame.to_numpy()";
+
   @Test
   void test() {
-    PythonCheckVerifier.verify("src/test/resources/checks/pandasDataFrameToNumpy.py", new PandasDataFrameToNumpyCheck());
+    PythonCheckVerifier.verify("src/test/resources/checks/pandasDataFrameToNumpy.py", check);
+  }
+
+  @Test
+  void quick_fix_test_1() {
+    final String non_compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    df = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    })\n" +
+      "\n" +
+      "    _ = df.values";
+    final String compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    df = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    })\n" +
+      "\n" +
+      "    _ = df.to_numpy()";
+    performVerification(non_compliant, compliant);
+  }
+
+  @Test
+  void quick_fix_test_2() {
+
+    final String non_compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    _ = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    }).values";
+    final String compliant = "def non_compliant_1(xx):\n" +
+      "    import pandas as pd\n" +
+      "\n" +
+      "    _ = pd.DataFrame({\n" +
+      "        'X': ['A', 'B', 'A', 'C'],\n" +
+      "        'Y': [10, 7, 12, 5]\n" +
+      "    }).to_numpy()";
+    performVerification(non_compliant, compliant);
   }
 
+  @Test
+  void quick_fix_test_3() {
```

Comment:
I share David's opinion that here, we're not really testing the quick fix but rather the fact that we raise an issue on the DataFrame obtained through `read_csv`. I don't think it's a problem to keep the test either, though.

If you want additional quick fix tests, generally you can keep simple issues but play with things like context / indentation. For instance, what happens to the quick fix if the `values` attribute access is followed by another call?

For instance, would `df.values.astype(str)` be quick fixed into `df.values.to_numpy().astype(str)`?